platform/run_image.go

```diff
@@ -100,3 +62,39 @@ func byRegistry(reg string, images []string, checkReadAccess CheckReadAccess, ke
 	}
 	return ""
 }
+
+// GetRunImageForExport TODO
+func GetRunImageForExport(inputs LifecycleInputs) (files.RunImageForExport, error) {
+	if inputs.PlatformAPI.LessThan("0.12") {
+		stackMD, err := files.ReadStack(inputs.StackPath, cmd.DefaultLogger)
+		if err != nil {
+			return files.RunImageForExport{}, err
+		}
+		return stackMD.RunImage, nil
+	}
+	runMD, err := files.ReadRun(inputs.RunPath, cmd.DefaultLogger)
+	if err != nil {
+		return files.RunImageForExport{}, err
+	}
+	if len(runMD.Images) == 0 {
+		return files.RunImageForExport{}, nil
+	}
+	analyzedMD, err := files.ReadAnalyzed(inputs.AnalyzedPath, cmd.DefaultLogger)
+	if err != nil {
+		return files.RunImageForExport{}, err
+	}
+	for _, runImage := range runMD.Images {
+		if runImage.Contains(analyzedMD.RunImageImage()) {
+			return runImage, nil
+		}
+	}
+	buildMD := &files.BuildMetadata{}
+	if err = files.DecodeBuildMetadata(launch.GetMetadataFilePath(inputs.LayersDir), inputs.PlatformAPI, buildMD); err != nil {
+		return files.RunImageForExport{}, err
+	}
+	if len(buildMD.Extensions) > 0 { // FIXME: try to know for sure if extensions were used to switch the run image
```

Comment:
how could you know for sure?

Reply:
Not for sure without some spec changes... but I would rather tackle this in tandem with https://github.com/buildpacks/rfcs/pull/285